pkg/jx/cmd/step_helm_apply.go

```diff
@@ -242,36 +242,47 @@ func (o *StepHelmApplyOptions) Run() error {
 func (o *StepHelmApplyOptions) applyTemplateOverrides(chartName string) error {
 	log.Infof("Applying chart overrides\n")
 	templateOverrides, err := filepath.Glob(chartName + "/../*/templates/*.yaml")
+	if err != nil {
+		return errors.Wrap(err, "Failed to find templates")
+	}
 	for _, overrideSrc := range templateOverrides {
-		if !strings.Contains(overrideSrc, "/env/") {
+		if !strings.Contains(overrideSrc, filepath.FromSlash("/env/")) {
 			data, err := ioutil.ReadFile(overrideSrc)
-			if err == nil {
-				writeTemplateParts := strings.Split(overrideSrc, string(os.PathSeparator))
-				depChartsDir := filepath.Join(chartName, "charts")
-				depChartName := writeTemplateParts[len(writeTemplateParts)-3]
-				templateName := writeTemplateParts[len(writeTemplateParts)-1]
-				depChartDir := filepath.Join(depChartsDir, depChartName)
-				// If the chart directory does not exist explode the tgz
-				if exists, err := util.DirExists(depChartDir); err == nil && !exists {
-					chartArchives, _ := filepath.Glob(filepath.Join(depChartsDir, depChartName+"*.tgz"))
-					if len(chartArchives) == 1 {
-						log.Infof("Exploding chart %s\n", chartArchives[0])
-						archiver.Unarchive(chartArchives[0], depChartsDir)
-						// Remove the unexploded chart
-						os.Remove(chartArchives[0])
+			if err != nil {
+				return errors.Wrapf(err,"read %s", overrideSrc)
+			}
+			writeTemplateParts := strings.Split(overrideSrc, string(os.PathSeparator))
+			depChartsDir := filepath.Join(chartName, "charts")
+			depChartName := writeTemplateParts[len(writeTemplateParts)-3]
+			templateName := writeTemplateParts[len(writeTemplateParts)-1]
+			depChartDir := filepath.Join(depChartsDir, depChartName)
+			// If the chart directory does not exist explode the tgz
+			exists, err := util.DirExists(depChartDir)
+			if err != nil {
+				return errors.Wrapf(err, "Checking directory %s", depChartDir)
+			}
+			if !exists {
+				chartArchives, _ := filepath.Glob(filepath.Join(depChartsDir, depChartName+"*.tgz"))
```

Comment:
this is the existing logic - but if it enters here it will not unpack the archive as it was expecting to do as the Glob will not match anything as the directory does not exist??!!!

Reply:
now I could just change the logic here - but the comment on [line 259](https://github.com/jenkins-x/jx/pull/3753/files#diff-ee8c207b832edc0919065b6959cbfdcbR259) implied that this should be unpacking something if the directory does not exist - so in that case the question becomes where is this thing that should be unpacked?

@jstrachan any ideas here?